tests/cli/test_buildspec.py

```diff
@@ -251,3 +252,10 @@ def test_buildspec_show():
             random.choice(string.ascii_letters) for i in range(10)
         )
         show_buildspecs(test_names=[random_testname], configuration=configuration)
+
+
+@pytest.mark.cli
+def test_buildspec_show_fail():
+    # test buildtest buildspec show-fail
+    with pytest.raises(BuildTestError):
+        show_failed_buildspecs(configuration)
```

Comment:
i think this regression test needs some work, i think if you want to catch the exception in following situation
1. Query some random test name that doesn't exist which should raise exception. 
2. Query a test that is NOT in `state=FAIL` so you will need to query all pass test and extract the first one as an argument to `show_failed_buildspecs` command.

Reply:
You should also consider having an optional argument to specify a list of test names that can be passed to `show_failed_buildspecs` when testing otherwise your method will extract all failed tests.

So something like `def show_failed_buildspecs(configuration, names=None, report_file=None)`


Then when you are trying to get failed tests it would be something like. 

```
failed_tests = names or results.get_failed_tests()
show_buildspecs(failed_tests, configuration)
```